services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -0,0 +1,533 @@
+     *
+     * @param attributeName Name of the attribute.
+     * @param type          {@link EnhancedType} of Type T.         
+     * @param <T>           Type T of List elements
+     * @return value of the specified attribute in the current document as a list of type T; or null if the
+     * attribute either doesn't exist or the attribute value is null.
+     */
+
+    <T> List<T> getList(String attributeName, EnhancedType<T> type);
+
+
+    /**
+     * Gets the List of values for the given attribute in the current document.
+     * @param attributeName Name of the attribute.
+     * @return value of the specified attribute in the current document as a list; or null if the
+     * attribute either doesn't exist or the attribute value is null.
+     */
+    List<?> getList(String attributeName);
+
+    /**
+     * Gets the Map with Key as String and values as type T for the given attribute in the current document.
+     * <p>Note that any numeric type of map is always canonicalized into {@link SdkNumber}, and therefore if <code>T</code>
+     * referred to a <code>Number</code> type, it would need to be <code>SdkNumber</code> to avoid a class cast exception.
+     * </p>
+     *
+     * @param attributeName Name of the attribute.
+     * @param type          {@link EnhancedType} of Type T.              
+     * @param <T>           Type T of List elements
+     * @return value of the specified attribute in the current document as a map of string-to-<code>T</code>'s; or null if the
+     * attribute either doesn't exist or the attribute value is null.
+     */
+    <T> Map<String, T> getMap(String attributeName, EnhancedType<T> type);
+
+    /**
+     * Convenience method to return the specified attribute in the current item as a (copy of) map of
+     * string-to-<code>SdkNumber</code>'s where T must be a subclass of <code>Number</code>; or null if the attribute doesn't
+     * exist.
+     *
+     * @param attributeName Name of the attribute.
+     * @param valueType     the specific number type of the value to be returned.
+     *                     Currently, the supported types are
+     *                     <ul>
+     *                      <li><code>Short</code></li>
+     *                      <li><code>Integer</code></li>
+     *                      <li><code>Long</code></li>
+     *                      <li><code>Float</code></li>
+     *                      <li><code>Double</code></li>
+     *                      <li><code>Number</code></li>
+     *                      <li><code>BigDecimal</code></li>
+     *                      <li><code>BigInteger</code></li>
+     *                     </ul>
+     * @return value of the specified attribute in the current item as a (copy of) map
+     */
+    <T extends Number> Map<String, T> getMapOfNumbers(String attributeName,
+                                                      Class<T> valueType);
```

Comment:
Do we need T to extend numbers, or can it just be `getMapOfType`? Do we even need this if they can use get with an enhanced attribute type of map?